src/modules/ProjectsModule/components/ProjectsList/ProjectsList.tsx

```diff
@@ -1,10 +1,179 @@
+
+  const [projectId, setProjectId] = useState('')
+  
+  const [showDelete, setShowDelete] = useState(false);
+  const handleCloseDelete = () => setShowDelete(false);
+  const handleShowDelete = (id) => {
+   setProjectId(id);
+   setShowDelete(true);   
+  }
+  
+  const getProjects = async(title) => {
+    try {
+      let response = await axios.get(`${baseUrl}/Project/manager`,
+        { headers: { Authorization: authorization } , params:{title} })
+      setProjectsList(response.data.data)
+   
+    } catch (error) {
+       toast.error(error.message, {
+        autoClose: 3000,
+        hideProgressBar: true,
+        pauseOnHover: false,
+      });
+     
+    }
+  }
+
+  const deleteProjectItem = async () => {
+    try {
+      let deletedItem = await axios.delete(
+        `${baseUrl}/Project/${projectId}`,
+        {
+          headers: { Authorization: `Bearer ${localStorage.getItem("token")}` },
+        }
+      );
+      toast.warning("Item Deleted Successfully", {
+        autoClose: 3000,
+        hideProgressBar: true,
+        pauseOnHover: false,
+      });
+      handleCloseDelete();
+       getProjects()
+    } catch (error:any) {
+      
+         toast.error(error.message, {
+        autoClose: 3000,
+        hideProgressBar: true,
+        pauseOnHover: false,
+      });
+    }
+  }
+
+    const formatDate = (dateString) => {
+    const date = new Date(dateString);
+    const month = date.getMonth() + 1;
+    const day = date.getDate();
+    const year = date.getFullYear();
+    return `${month}/${day}/${year}`;
+  };
+
+  const handleUpdate = (id,title,description) => {
+    setMode('update')
+    setTitle(title)
+    setDescription(description)
+    navigate(`/dashboard/projects-data/${id}`)
+  }
+
+  const getNameValue = (e) => {
+    getProjects(e.target.value)
+  }
+
+  useEffect(() => {
+  getProjects('')
+  }, [])
 
-const ProjectsList = () => {
   return (
-    <>
-     ProjectsList 
-    </>
-  )
+<>
+        <Modal show={showDelete} onHide={handleCloseDelete}>
+        <Modal.Header closeButton></Modal.Header>
+        <Modal.Body className="my-2 px-5 ">
+      <div className="text-center">
+      <img className="w-25" src={deleteTrash} alt="" />
+      <h4 className="py-3">Delete This Project ?</h4>
+      <p className="text-muted">
+        are you sure you want to delete this item ? if you are sure just click
+        on delete it
+      </p>
+      </div>
+          
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="danger" onClick={deleteProjectItem}>
+            Delete This Item
+          </Button>
+        </Modal.Footer>
+      </Modal>
+
+    <div className='projectsList vh-100'>
+      <div className="projects-container bg-white px-3 d-flex justify-content-between py-4">
+        <h3>Projects</h3>
+        <Link onClick={setMode('create')} className='px-4 rounded-5' to={'/dashboard/projects-data'}> <i className='fa fa-plus pe-3'></i> Add New Project</Link>
```

Comment:
We shouldn't be giving onClick to links. If we are doing something else beside navigation, then we should use a button and have a programmatic navigation using useNavigate hook.